signup.js

```diff
@@ -0,0 +1,82 @@
+const signupBtn = document.getElementById('signup-button');
+const form = document.getElementById('Form'); 
+const emailField = document.querySelector(".email-field");
+const emailInput = document.querySelector(".email");
+const passInput = document.getElementById('password');
+const eyeIcons = document.querySelectorAll(".visibility_off"); 
+const passField = document.querySelector(".create-password");
+const cPassInput = document.getElementById('password-repeat');
+const confirmError = document.querySelector(".confirm-error");
+
+eyeIcons.forEach(eyeIcon => {
+  eyeIcon.addEventListener("click", () => {
+    const passwordInput = eyeIcon.parentElement.querySelector("input");
+
+    if (passwordInput.type === "password") {
+      passwordInput.type = "text"; 
+      eyeIcon.src = "image/visibility_on_btn.png";
+    } else {
+      passwordInput.type = "password"; 
+      eyeIcon.src = "image/visibility_off_btn.png"; 
+    }
+  });
+});
+
+function checkEmail() {
+  const emailPattern = /^[^ ]+@[^ ]+\.[a-z]{2,3}$/;
+
+  if (!emailInput.value.match(emailPattern)) {
+    emailField.classList.add("invalid"); 
+    emailField.querySelector(".email-error").style.display = "block";
+  } else {
+    emailField.classList.remove("invalid"); 
+    emailField.querySelector(".email-error").style.display = "none";
+  }
+}
+
+function createPass() {
+  const passPattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+
+  if (!passInput.value.match(passPattern) || passInput.value.length < 8) {
+    passInput.classList.add("invalid");
+    passField.querySelector(".password-error").style.display = "block";
+  } else {
+    passInput.classList.remove("invalid");
+    passField.querySelector(".password-error").style.display = "none";
+  }
+}
+
+function checkPasswordMatch() {
+  const passValue = passInput.value;
+
+  if (passValue !== cPassInput.value) {
+    cPassInput.classList.add("invalid");
+    confirmError.style.display = "block";
+  } else {
+    cPassInput.classList.remove("invalid");
+    confirmError.style.display = "none";
+  }
+}
+
+form.addEventListener("submit", function(e) {
+  e.preventDefault();
+  checkEmail(); 
+  createPass();
+  checkPasswordMatch();
+});
```

Comment:
내용들을 읽어보니 submit 이벤트에 1)유효성검사 2) 에러메시지 보여주기 를 하는 걸로 보이네요.
다만 이 부분을 봤을 때 그 동작이 드러나면 좋겠습니다
지금은 함수이름들로 봤을 떄, 3개의 함수들을 개별로 순서대로 호출하는 걸로 보입니다.
유효성검증와 UI변화를 하나의 함수가 아니라 별도 함수로 분리하는 방향으로, 즉 함수가 가진 관심사에 따라 별도 함수로 나뉘는 방향으로 개발하시면 좋겠습니다

```suggestion
form.addEventListener("submit", function(e) {
  e.preventDefault();
  if(!validateEmail()) { // 검증로직에 관한 함수 
    showEmailErrorMessage() // ui 변화에 관한 함수
  }
  // 패스워드 검증 + 에러메시지 표현
  // 패스워드 일치 검증 + 에러메시지 표현
});
```